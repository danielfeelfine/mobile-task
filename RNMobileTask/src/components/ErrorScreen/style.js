import { StyleSheet } from 'react-native'
import { colors } from '@styles/base'

const style = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 38,
    paddingHorizontal: 16,
    alignItems: 'center',
    backgroundColor: colors.Light,
  },
  title: {
    marginTop: 6,
    marginBottom: 12,
    textAlign: 'center',
    fontWeight: '800',
    fontSize: 18,
    lineHeight: 22,
    color: colors.Danger,
  },
  message: {
    marginVertical: 4,
    paddingHorizontal: 12,
    textAlign: 'center',
    fontWeight: '300',
    fontSize: 12,
    lineHeight: 16,
    color: colors.Dark,
  },
  subtitle: {
    marginVertical: 4,
    fontWeight: '600',
    fontSize: 12,
    color: colors.Dark,
  }
})

export default style
