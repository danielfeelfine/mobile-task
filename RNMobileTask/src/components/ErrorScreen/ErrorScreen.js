// @flow
import React from 'react'
import { Text, View } from 'react-native'
import style from './style'

type Props = {
  message: string
}

const ErrorScreen = (props: Props) => {
  return (
    <View style={style.container}>
      <Text style={style.title}>Sorry! :'(</Text>
      <Text style={style.subtitle}>An error occurred:</Text>
      <Text style={style.message}>{props.message}</Text>
    </View>
  )
}

export default ErrorScreen
