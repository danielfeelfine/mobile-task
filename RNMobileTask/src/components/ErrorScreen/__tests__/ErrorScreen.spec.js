import React from 'react'
import { shallow } from 'enzyme'
import ErrorScreen from './../ErrorScreen'


describe('ErrorScreen component', () => {

  it('renders correctly', () => {
    const props = {
      message: 'Network request failed'
    }

    const wrapper = shallow(
      <ErrorScreen message={props.message} />
    )
  
    expect(wrapper).toMatchSnapshot()
  })

})
