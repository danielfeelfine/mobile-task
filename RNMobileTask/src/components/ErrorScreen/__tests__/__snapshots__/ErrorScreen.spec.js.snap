// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorScreen
    message="Network request failed"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FC1818",
              "fontSize": 18,
              "fontWeight": "800",
              "lineHeight": 22,
              "marginBottom": 12,
              "marginTop": 6,
              "textAlign": "center",
            }
          }
        >
          Sorry! :'(
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "fontSize": 12,
              "fontWeight": "600",
              "marginVertical": 4,
            }
          }
        >
          An error occurred:
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 16,
              "marginVertical": 4,
              "paddingHorizontal": 12,
              "textAlign": "center",
            }
          }
        >
          Network request failed
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 38,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Sorry! :'(",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#FC1818",
            "fontSize": 18,
            "fontWeight": "800",
            "lineHeight": 22,
            "marginBottom": 12,
            "marginTop": 6,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Sorry! :'(",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "An error occurred:",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "fontSize": 12,
            "fontWeight": "600",
            "marginVertical": 4,
          },
        },
        "ref": null,
        "rendered": "An error occurred:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Network request failed",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 16,
            "marginVertical": 4,
            "paddingHorizontal": 12,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Network request failed",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FC1818",
                "fontSize": 18,
                "fontWeight": "800",
                "lineHeight": 22,
                "marginBottom": 12,
                "marginTop": 6,
                "textAlign": "center",
              }
            }
          >
            Sorry! :'(
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "600",
                "marginVertical": 4,
              }
            }
          >
            An error occurred:
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 16,
                "marginVertical": 4,
                "paddingHorizontal": 12,
                "textAlign": "center",
              }
            }
          >
            Network request failed
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 38,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Sorry! :'(",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#FC1818",
              "fontSize": 18,
              "fontWeight": "800",
              "lineHeight": 22,
              "marginBottom": 12,
              "marginTop": 6,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Sorry! :'(",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "An error occurred:",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "fontSize": 12,
              "fontWeight": "600",
              "marginVertical": 4,
            },
          },
          "ref": null,
          "rendered": "An error occurred:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Network request failed",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 16,
              "marginVertical": 4,
              "paddingHorizontal": 12,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Network request failed",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
